### PR TITLE
tests: lib: use new ztest API for at_custom_cmd and gcf_sms tests

### PR DESCRIPTION
**Pull request by eivindj-nordic**

### CODE REVIEW

**Review by gopiotr — COMMENTED** [review]
LGTM. Some minor suggestions.

File: tests/lib/gcf_sms/src/main.c [hunk 1]
```diff
@@ -14,210 +14,236 @@
...
-#define CSMS_3_CMD "AT+CSMS?"
-#define CSMS_3_RES "+CSMS: 0,1,1,0\r\nOK\r\n"
+ZTEST(gcf_sms, test_gcf_sms_cpms)
+{
+	int err;
 
-	err = nrf_modem_at_cmd(response, sizeof(response), CSMS_3_CMD);
+	err = nrf_modem_at_cmd(response, sizeof(response), CMD_CPMS_E_TA);
+
+	printk("res: %s\n", response);
```
[the remark this answers is not in the document]
Comment (eivindj-nordic) [reply tests/lib/gcf_sms/src/main.c:99]: Yes, thanks :)

**Review by gopiotr — APPROVED** [review]
Thank you a lot for your work :) From Ztest point of view it looks okay 👍

**Review by lemrey — APPROVED** [review]

**Comment by lemrey** [comment]
This only affects tests, no need for changelog entry.

**Pull request merged**